### PR TITLE
dist: Removed unused code and fix travis build

### PR DESCRIPTION
**Pull request by ChrisAlderson**
Changes to `index.js`:
 - Remove use or arguments in _makeCached
 - Use short-hand property in parseArgs
 - Remove unused methods _parseArgs and _processArgs 

Changes to `.travis.yml`
 - dist: Only test on lts and latest version of node
 - dist: Use default test script from travis
 - dist: Use default install script from travis
 - dist: Do not cache dependencies on travis 

Got rid of `eslint` in favor of `standardjs` because you are using it in other `butter` related projects too.

### CODE REVIEW

File: tests/index.spec.js [hunk 1]
```diff
@@ -234,11 +234,11 @@ describe('Provider Instance', () => {
   describe('resolveStream', () => {
     it('should parse a string to an object', () => {
       const uri = 'ProviderName?key1=["value1"]&key2=value2'
-      const parsed = provider._parseArgs(uri)
+      const parsed = Provider.parseArgs(uri)
```
Comment (xaiki) [inline tests/index.spec.js]: this change makes it that we don't pass the config anymore

File: index.js [hunk 2]
```diff
@@ -112,34 +109,24 @@ class Provider {
...
           // Delete the cached result if we get an error so retry will work
-          memoizedMethod.delete(self.filters)
+          memoizedMethod.delete(this.filters)
           return Promise.reject(err)
         })
     }
   }
 
-  _parseArgs(argString) {
-    return Provider.parseArgs(argString, this.config.argTypes)
```
Comment (xaiki) [inline index.js]: _parseArgs and parseArgs are not equivalent, one gets argTypes from config

File: index.js [hunk 3]
```diff
@@ -112,34 +109,24 @@ class Provider {
...
-  }
-
-  _processArgs(argString) {
-    processArgs(argString, this.config)
```
Comment (xaiki) [inline index.js]: same here for _processArgs

File: package-lock.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
 {
```
Comment (xaiki) [inline package-lock.json]: this file should probably not be in git

File: .travis.yml [hunk 5]
```diff
@@ -15,9 +15,5 @@ install:
   - npm i -g npm@latest
   - npm i
 
-script:
-  - npm run lint
-  - npm run test
-
```
Comment (xaiki) [inline .travis.yml]: i feel this might bite us later, what if travis silently stops linting ? or changes the test target ?

File: .travis.yml [hunk 6]
```diff
@@ -7,9 +7,5 @@ node_js:
   - 8
   - 10
 
-cache:
-  directories:
-    - node_modules
-
```
Comment (xaiki) [inline .travis.yml]: i guess caching is being nice to travis, and speeds up our builds,
why not be nice and fast ?

File: .eslintrc.js [hunk 7]
```diff
@@ -1,21 +0,0 @@
...
-      switches: 'never',
-      classes: 'always'
-    }],
-    'space-before-function-paren': ['error', {
-      anonymous: 'always',
-      named: 'never',
-      asyncArrow: 'always'
-    }]
-  }
-}
```
Comment (xaiki) [inline .eslintrc.js]: commit message is wrong, should say: 'remove eslint' should be squashed into the commit that removes it from package.json too

File: .travis.yml (hunk 6 above)
Comment (ChrisAlderson) [reply .travis.yml]: I agree, but this is more a `package-lock.json` problem. npm can be a bit finicky with installing dependencies and using the `package-lock.json` file. Almost a year after npm introduced the `package-lock.json` file it's still a very common thing to do when having dependency issues, to `rm -rf node_modules package-lock.json`.

Speed doesn't really seem to be an issue anyway, this package is fairly small and the build for this PR took just over 1 minute and if we look at the builds from `master` they take about 1 and a half minute.

Hence removing the cache just ensures all the dependencies are installed cleanly and apparently it's faster.

File: index.js (hunk 2 above)
Comment (ChrisAlderson) [reply index.js]: Yet it is a private method which is never used inside the `Provider` class, so this just leaves me confused now.

File: package-lock.json (hunk 4 above)
Comment (ChrisAlderson) [reply package-lock.json]: How unreliable it sometimes it, npm says it's supposed to be in git. https://docs.npmjs.com/files/package-lock.json#description

Comment (xaiki) [reply package-lock.json]: in this repo it's probably ok,
but in our other repos that depend heavily on one another it just put huge chunks in diffs each time we update a dep.

File: index.js (hunk 2 above)
Comment (xaiki) [reply index.js]: it's used in tests

Comment (xaiki) [reply index.js]: it ... is used in the class.

File: .travis.yml (hunk 6 above)
Comment (xaiki) [reply .travis.yml]: following that logic shouldn't we remove package-lock.json too ?

**Pull request merged**